python/dgl/data/gnn_benckmark.py

```diff
@@ -16,24 +20,40 @@ def eliminate_self_loops(A):
     return A
 
 
-class GNNBenchmarkDataset(object):
-    """Base Class for GNN Benchmark dataset from https://github.com/shchur/gnn-benchmark#datasets"""
-    _url = {}
+class GNNBenchmarkDataset(DGLBuiltinDataset):
+    r"""Base Class for GNN Benchmark dataset from https://github.com/shchur/gnn-benchmark#datasets"""
+    _url = None
 
-    def __init__(self, name):
-        assert name.lower() in self._url, "Name not valid"
-        self.dir = get_download_dir()
-        self.path = os.path.join(
-            self.dir, 'gnn_benckmark', self._url[name.lower()].split('/')[-1])
-        self._name = name
-        g = self.load_npz(self.path)
+    def __init__(self, name, force_reload=False):
+        super(GNNBenchmarkDataset, self).__init__(name=name, url=self._url, force_reload=force_reload)
+
+    def process(self, root_path):
+        g = self._load_npz(root_path + '.npz')
+        self.graph = g
         self.data = [g]
 
-    def _download(self):
-        download(self._url[self._name.lower()], path=self.path)
+    def has_cache(self):
+        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
+        if os.path.exists(graph_path):
+            return True
+        return False
+
+    def save(self):
+        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
+        save_graphs(graph_path, self.graph)
 
-    @retry_method_with_fix(_download)
-    def load_npz(self, file_name):
+    def load(self):
+        graph_path = os.path.join(self.save_path, 'dgl_graph.bin')
```

Comment:
Sorry for saying this late. Can we put the save_path and the filenames into the class attribute instead of instance attribute? (Just like _url). This makes it easy to identify which files are used for cache/saving. And for the root_path or other paths, are they standardized across all DGL datasets?